File: tool/measurementDriver/src/ch/ethz/ruediste/roofline/dom/WorkloadStartEventPredicate.java
package ch.ethz.ruediste.roofline.dom;

public class WorkloadStartEventPredicate extends
		WorkloadStartEventPredicateData {

	private Workload workload;

	public WorkloadStartEventPredicate() {
		super();
	}

	public WorkloadStartEventPredicate(Workload workload) {
		this();
		this.workload = workload;
	}

	public Workload getWorkload() {
		return workload;
	}

	public void setWorkload(Workload workload) {
		this.workload = workload;
	}

	@Override
	public int getWorkloadId() {
		return workload.getId();
	}
}

File: tool/measuringCore/src/sharedEntities/eventPredicates/WorkloadStartEventPredicate.cpp
/*
 * WorkloadStartRule.cpp
 *
 *  Created on: Feb 9, 2012
 *      Author: ruedi
 */

#include "WorkloadStartEventPredicate.h"

#include "baseClasses/events/WorkloadStartEvent.h"
#include "sharedEntities/Workload.h"

#include <typeinfo>

bool WorkloadStartEventPredicate::doesMatch(EventBase *event)
{
	// check if it is a start event
	if (typeid(*event)==typeid(WorkloadStartEvent)){
		WorkloadStartEvent *startEvent=(WorkloadStartEvent*)event;

		// check the workload id
		if (getWorkload()==NULL || startEvent->getWorkloadId()==getWorkload()->getId()){
			return true;
		}
	}
	return false;
}

